internal/pkg/deploy/cloudformation/stack/task.go

```diff
@@ -0,0 +1,117 @@
+package stack
+
+import (
+	"fmt"
+	"strconv"
+
+	"github.com/aws/copilot-cli/internal/pkg/deploy"
+	"github.com/aws/copilot-cli/internal/pkg/template"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/cloudformation"
+)
+
+const (
+	taskTemplatePath = "task/cf.yml"
+
+	taskNameParamKey         = "TaskName"
+	taskCPUParamKey          = "TaskCPU"
+	taskMemoryParamKey       = "TaskMemory"
+	taskLogRetentionParamKey = "LogRetention"
+
+	taskContainerImageParamKey = "ContainerImage"
+	taskTaskRoleParamKey       = "TaskRole"
+	taskCommandParamKey        = "Command"
+
+	taskLogRetention = "1"
+)
+
+type taskStackConfig struct {
+	Name string
+
+	Cpu    int
+	Memory int
+
+	ImageURL string
+	TaskRole string
+	Command  string
+	EnvVars map[string]string
+
+	parser template.ReadParser
+}
+
+// NewTaskStackConfig sets up a struct that provides stack configurations for CloudFormation
+// to deploy the task resources stack.
+func NewTaskStackConfig(taskOpts *deploy.CreateTaskResourcesInput) *taskStackConfig {
+	return &taskStackConfig{
+		Name:   taskOpts.Name,
```

Comment:
How do you like `TaskGroupName`? Because this `Name` field is passed to the `TaskName` parameter in the template, which is then referenced not only in `Name` of the Container definition and `Family`, but also LogGroupName, RepositoryName (in ECR)